src/modules/verifiableCredentials/verifiableCredentialsBase.service.ts

```diff
@@ -32,67 +58,69 @@ export abstract class VerifiableCredentialsServiceBase {
     throw new Error('Not implemented');
   }
 
-  public async verifyVerifiableCredential<T>(
-    vc: SignedVerifiableCredential<T>,
-    options?: VerifyVerifiableCredentialOptions
-  ) {
-    const verifyResultsString = await this.verifyCredential(
-      JSON.stringify(vc),
-      JSON.stringify(options || {})
-    );
-
-    const verifyResults = JSON.parse(
-      verifyResultsString
-    ) as VerifyVerifiableCredentialResults;
+  private generateEIP712Domain(domain: TypedDataDomain) {
+    const types = {
+      name: 'string',
+      version: 'string',
+      chainId: 'uint256',
+      verifyingContract: 'address',
+      salt: 'bytes32',
+    };
 
-    if (verifyResults.errors.length) {
-      throw new Error(
-        `Verifiable Credential is invalid, errors: ${verifyResults.errors.join(
-          ', '
-        )}`
-      );
-    }
+    const keys = Object.keys(domain);
+    return keys.map((key) => ({ type: types[key], name: key }));
+  }
 
-    return true;
+  // TODO: Host EWF VC Context and Vocab
+  private createCredentialObject(
+    params: CreateVerifiableCredentialParams
+  ): VerifiableCredential<EWFCredentialSubject> {
+    const credential = {
+      '@context': ['https://www.w3.org/2018/credentials/v1'],
+      id: 'urn:uuid:' + uuid(),
+      type: ['VerifiableCredential', 'EWFRole'],
+      issuer: this._signerService.didHex,
+      issuanceDate: new Date().toISOString(),
+      credentialSubject: {
+        role: {
+          namespace: params.namespace,
+          version: params.version,
+        },
+        issuerFields: params.issuerFields
+          ? [
+              ...params.issuerFields.map((field) => ({
+                ...field,
+                value: field.value.toString(),
+              })),
+            ]
+          : [],
+        id: params.subject,
+      },
+    };
+    return credential;
   }
 
-  public async signVerifiableCredential<T>(
-    credential: VerifiableCredential<T>,
+  // TODO: Check if issuer is allowed to issue given credential
```

Comment:
I would remove this TBH. IMO, this would imply checking the role definition or something the sort and that would couple this code to a governance framework.
```suggestion
```

Reply:
So verifying will need to check it?

Reply:
The issuer might also check it, but I don't think it should be done in this location. IMO, as the name is `signVerificableCredential`, function should only do one thing (following the Single Responsibility Principle), which is sign the credential.